test/e2e/route_advertisements.go

```diff
@@ -1074,3 +1669,519 @@ func generateL2Subnets(v4, v6 string) udnv1.DualStackCIDRs {
+	udn                   networkType = "UDN"
+	cudn                  networkType = "CUDN"
+	cudnAdvertised        networkType = "CUDN_ADVERTISED"
+	cudnAdvertisedVRFLite networkType = "CUDN_ADVERTISED_VRFLITE"
+)
+
+// createNamespaceWithPrimaryNetworkOfType helper function configures a
+// namespace, a optional(C)UDN and an optional RouteAdvertisements as determined
+// by `networkType` argument. The RouteAdvertisements is aligned with the
+// configuration done with `runBGPNetworkAndServer` for VRF-Lite scenarios.
+func createNamespaceWithPrimaryNetworkOfType(
+	f *framework.Framework,
+	ictx infraapi.Context,
+	test, name string,
+	networkType networkType,
+	networkSpec *udnv1.NetworkSpec,
+) (*corev1.Namespace, error) {
+	// define some configuration based on the type of namespace/network/advertisement
+	var targetVRF string
+	var networkLabels map[string]string
+	var frrConfigurationLabels map[string]string
+	switch networkType {
+	case cudnAdvertised:
+		networkLabels = map[string]string{"advertise": name}
+		frrConfigurationLabels = map[string]string{"name": "receive-all"}
+	case cudnAdvertisedVRFLite:
+		targetVRF = name
+		networkLabels = map[string]string{"advertise": name}
+		frrConfigurationLabels = map[string]string{"network": name}
+	}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"e2e-framework": test,
+			},
+		},
+	}
+	if networkType != defaultNetwork {
+		namespace.Labels[RequiredUDNNamespaceLabel] = ""
+	}
+	namespace, err := f.ClientSet.CoreV1().Namespaces().Create(
+		context.Background(),
+		namespace,
+		metav1.CreateOptions{},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create namespace: %w", err)
+	}
+	ictx.AddCleanUpFn(func() error {
+		return f.ClientSet.CoreV1().Namespaces().Delete(context.Background(), namespace.Name, metav1.DeleteOptions{})
+	})
+
+	// just creating a namespace with default network, return
+	if networkType == defaultNetwork {
+		return namespace, nil
+	}
+
+	err = createUserDefinedNetwork(
+		f,
+		ictx,
+		namespace,
+		name,
+		networkType != udn,
+		networkSpec,
+		networkLabels,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create primary network: %w", err)
+	}
+
+	// not advertised, return
+	if networkType == udn || networkType == cudn {
+		return namespace, nil
+	}
+
+	err = createRouteAdvertisements(
+		f,
+		ictx,
+		name,
+		targetVRF,
+		networkLabels,
+		frrConfigurationLabels,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create primary network: %w", err)
+	}
+
+	return namespace, nil
+}
+
+func createUserDefinedNetwork(
+	f *framework.Framework,
+	ictx infraapi.Context,
+	namespace *corev1.Namespace,
+	name string,
+	cudnType bool,
+	networkSpec *udnv1.NetworkSpec,
+	networkLabels map[string]string,
+) error {
+	var gvr schema.GroupVersionResource
+	var gvk schema.GroupVersionKind
+	var obj runtime.Object
+	var client dynamic.ResourceInterface
+	switch cudnType {
```

Comment:
why not use an `if cudnType`? :sweat_smile:

Reply:
I guess I am a fan of switch statement and their readability vs if/else but I really meant to do 
```
switch {
case cudnType == true:
default:
}
```